### DISCUSSION TITLE
airflow worker OOM

### DISCUSSION THREAD

**Discussion by Lee2532** [apache/airflow#35920] [Q&A]
when i build celery concurrency =100 then worker is OOM noting work task only building

concurrency  20  then not OOM
Does it basically eat up memory based on the number of concurrency?

Resource limit gave me 8 GB of memory

airflow version : 2.7.3
python version : 3.9

**Comment by potiuk** [comment]
> Does it basically eat up memory based on the number of concurrency?

Yes it spanws 100 processes that are waiting to pick up tasks. Celery works as "pool" of processes processing tasks. If you have more questions you can ask in calary forums or read their docs on architecture (highly recommended).

**Reply by Lee2532** [reply]
Thank you. I'll look for related documents
